### PR TITLE
fix(connector): fix null pointer in connector

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

As title.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/jni_core/src/lib.rs [hunk 1]
```diff
@@ -1024,6 +1024,9 @@ extern "system" fn Java_com_risingwave_java_binding_Binding_sendCdcSourceErrorTo
     msg: JString<'a>,
 ) -> jboolean {
     execute_and_catch(env, move |env| {
+        if msg.is_null() {
```
Comment (wenym1) [inline src/jni_core/src/lib.rs:1027]: How about doing this when the following `get_string` return with err? We can also print a warning log there.

File: src/jni_core/src/lib.rs [hunk 2]
```diff
@@ -1129,6 +1132,9 @@ pub extern "system" fn Java_com_risingwave_java_binding_Binding_sendSinkWriterEr
...
```
Comment (wenym1) [inline src/jni_core/src/lib.rs:1135]: ditto.

And shall we check all callers to this method to ensure that the msg passed to here must not be null?

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzCdcEngineRunner.java [hunk 3]
```diff
@@ -86,7 +86,9 @@ public static CdcEngineRunner create(DbzConnectorConfig config, long channelPtr)
                                             message,
                                             error);
                                     String errorMsg =
-                                            (error != null ? error.getMessage() : message);
+                                            (error != null && error.getMessage() != null
```
Comment (wenym1) [inline java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzCdcEngineRunner.java:89]: What does the message mean here? Is it the error message, or the message that failed to send?

Comment (StrikeW) [reply java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzCdcEngineRunner.java:89]: The arg `message` is passed from the Debezium engine in the callback.
```java
/**
     * A callback function to be notified when the connector completes.
     */
    interface CompletionCallback {
        /**
         * Handle the completion of the embedded connector engine.
         *
         * @param success {@code true} if the connector completed normally, or {@code false} if the connector produced an error
         *            that prevented startup or premature termination.
         * @param message the completion message; never null
         * @param error the error, or null if there was no exception
         */
        void handle(boolean success, String message, Throwable error);
    }
```

File: src/jni_core/src/lib.rs (hunk 1 above)
Comment (StrikeW) [reply src/jni_core/src/lib.rs]: refined.

File: src/jni_core/src/lib.rs (hunk 2 above)
Comment (StrikeW) [reply src/jni_core/src/lib.rs]: I have fixed passing msg that could be null in the caller.

**Review by wenym1 — APPROVED** [review]

**Pull request merged**